Document volume workflow

Ebaneck opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->
'volume', 'documentation'

**Context**: 

See #1419 

**Summary**:

- Document how to create and delete volume objects using the CLI and GUI

**Acceptance criteria**: 

- Provide Clear, simple and concise steps on how to perform each operation.
- A reader should be able to use the documentation correctly.
- CI builds should obviously pass.



---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1419

Ebaneck committed:
doc/volume: Add storageclass creation doc

Ebaneck committed:
doc/volume: Add volume creation & deletion doc via cli

bert-e commented:
# Hello ebaneck,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality

Ebaneck committed:
doc/volume: Add volume creation via GUI

slaperche-scality commented on docs/operation/index.rst:
```diff
@@ -14,3 +14,4 @@ do not have a working MetalK8s_ setup.
    upgrade
    downgrade
    changing_node_hostname
+   volume_workflow/index
```
s/volume_workflow/volume_management/

slaperche-scality commented on docs/operation/volume_workflow/index.rst:
```diff
@@ -0,0 +1,16 @@
+Volume Management Workflow
```
can remove `Workflow` I think

slaperche-scality commented on docs/operation/volume_workflow/index.rst:
```diff
@@ -0,0 +1,16 @@
+Volume Management Workflow
+==========================
+This section highlights **MetalK8s Volume Management Workflow** which entails
```
ditto, remove `Worflow`

s/entails/covers/

slaperche-scality commented on docs/operation/volume_workflow/index.rst:
```diff
@@ -0,0 +1,16 @@
+Volume Management Workflow
+==========================
+This section highlights **MetalK8s Volume Management Workflow** which entails
+volume creation, volume deletion and volume provisioning neccessary for use in
```
`volume creation and volume deletion`, I don't think we cover ` volume provisioning neccessary for use in persistent data storage within a MetalK8s Cluster.`

slaperche-scality commented on docs/operation/volume_workflow/index.rst:
```diff
@@ -0,0 +1,16 @@
+Volume Management Workflow
+==========================
```
keep an empty line after `=====`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+StorageClass Creation
+=====================
```
keep an empty line after `=====`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+=====================
+**MetalK8s** makes use of StorageClasses in other to provision and use
+persistent volumes.
+This section hightlights how to create a Storageclass using the **CLI** only.
```
remove `only`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+StorageClass Creation
+=====================
+**MetalK8s** makes use of StorageClasses in other to provision and use
```
`**MetalK8s** uses **StorageClass** to configure how **Volume** are formatted and mounted`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+persistent volumes.
+This section hightlights how to create a Storageclass using the **CLI** only.
+
+Volume objects require a storageclass object to function properly. To create a
```
Can be removed I think, feels a bit redundant with the first parapgraph

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+Volume objects require a storageclass object to function properly. To create a
+storageclass object, perform the following steps.
+
+#. Create a storage class defintion.
```
`Create a StorageClass manifest`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+  .. tip::
+
+      Always set **WaitForFirstConsumer** as **volumeBindingMode** so as to
+      delay the binding and provisioning of a PersistentVolume until a Pod
```
`s/PersistentVolume/**Volume**/`, `s/Pod/**Pod**/`

Let's not mention PV in our doc, we want to hide this concept to our users.

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+      Always set **WaitForFirstConsumer** as **volumeBindingMode** so as to
+      delay the binding and provisioning of a PersistentVolume until a Pod
+      using the PersistentVolumeClaim is created.
```
`s/PersistentVolumeClaim/**PersistentVolumeClaim**/`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+      delay the binding and provisioning of a PersistentVolume until a Pod
+      using the PersistentVolumeClaim is created.
+
+#. Apply the storageclass.
```
`Create the StorageClass`

Ebaneck replied:
Well, i wanted this to follow with the command used.. since we are using a `kubectl apply` instead of a `kubectl create`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+      root@bootstrap $ kubectl apply -f storageclass.yml
+
+#. Verify that the storageclass was created.
```
`Check that the StorageClass was created`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+   .. code-block:: shell
+
+      root@bootstrap $ kubectl get storageclass metalk8s-demo-storageclass
```
```
      root@bootstrap $ kubectl get storageclass metalk8s-demo-storageclass
      NAME                         PROVISIONER                    AGE
      metalk8s-demo-storageclass   kubernetes.io/no-provisioner   2s
```

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+#. Create a storage class defintion.
+
+   Save the storage class definition example file as **storageclass.yaml**.
```
`You can define a new **StorageClass** using the following template:`

slaperche-scality commented on docs/operation/volume_workflow/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+         fsType: <filesystem_type>
+         mkfsOptions: '["-m", "0"]'
+
+  .. tip::
```
I wouldn't use a `tip` for that, as it's mandatory.

I would also document what the user can customize: `mountOptions`, `fsType` and `mkfsOptions`.

e.g.:

```
You can use:
* **mountOptions** to specify how the volume should be mounted
* **fsType** to specify the filesystem to use on the volume
* **mkfsOptions** to specify how the volume should be formatted (not that the options are passed as a JSON-encoded string).

**volumeBindingMode** should always be set to **WaitForFirstConsumer** in order to have properly working **Volume**.
```

slaperche-scality reviewed (changes requested):
First commit reviewed

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+Volume Workflow CLI Operations
```
Volume Management using the CLI

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+Volume Workflow CLI Operations
+==============================
```
Skip line after `=====`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+Volume Workflow CLI Operations
+==============================
+To use persistent volumes within a MetalK8s cluster, a volume object needs
```
To use persistent storage in a MetalK8s cluster, one needs to create **Volume** objects.
In order to create **Volume** you need to have **StorageClass** objects registered in your cluster (see [here insert a link to the relevant section of the doc]).

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+To use persistent volumes within a MetalK8s cluster, a volume object needs
+to be created. Volume objects depend on StorageClasses to function properly.
+
+Volume Creation CLI
```
`Volume Creation`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+to be created. Volume objects depend on StorageClasses to function properly.
+
+Volume Creation CLI
+-------------------
```
skip line

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+This section highlights how to create volume objects in a MetalK8s cluster
+using the **CLI**.
+
+.. important::
```
No, not necessarily: we already provide a **StorageClass* so one don't have to create one before creation a **Volume**

Ebaneck replied:
Not sure everyone will be using the `metalk8s-prometheus` storageclass for every new volume created.

slaperche-scality replied:
Of course not,
But it's also likely that solution such as Hyperdrive & Zenko comes with their own StorageClass so creating one by hand is not mandatory.

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+Volume Creation CLI
+-------------------
+This section highlights how to create volume objects in a MetalK8s cluster
```
This section describes how to create a **Volume** from the CI.

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+   you have completed the storageclass creation procedure listed in
+   :doc:`/operation/volume_workflow/storageclass_creation`
+
+#. Create a sample volume.
```
Create a Volume manifest

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+#. Create a sample volume.
+
+   Save the volume definition file as **volume.yml**.
```
`You can define a new **Volume** using the following template:`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+         storageClassName: <storageclass_name>
+         rawBlockDevice:
+           devicePath: <device_path>
+
```
Add a paragraph explaining the field that needs to be filled:

```
Here you can set:

* `name`: the name of your volume, must be unique
* `nodeName`: the name of the node where the volume will be located.
* `storageClassName`: the **StorageClass** to use
* `devicePath`: path to the device (on the node).
```

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+         rawBlockDevice:
+           devicePath: <device_path>
+
+#. Apply the volume.
```
Create the volume

Title don't have a period, they're not sentence.

Ebaneck replied:
Hmmm, this was not a title... I see it as a number `listing`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+   .. code-block:: shell
+
+       root@bootstrap $ kubectl get volume demo-volume
```
```
       root@bootstrap $ kubectl get volume demo-volume
       NAME          NODE        STORAGECLASS
       demo-volume   bootstrap   metalk8s-demo-storageclass
```

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+   .. code-block:: shell
+
+       root@bootstrap $ kubectl get volume <volume_name>
```
```
       root@bootstrap $ kubectl get volume <volume_name>
       NAME          NODE        STORAGECLASS
       <volume_name>   bootstrap   metalk8s-demo-storageclass
```

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+      root@bootstrap $ kubectl apply -f volume.yml
+
+
+#. Verify that the volume was created.
```
`Check that the **Volume** was created`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+       NAME          NODE        STORAGECLASS
+       demo-volume   bootstrap   metalk8s-demo-storageclass
+
+Volume Deletion CLI
```
`Volume Deletion`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+       demo-volume   bootstrap   metalk8s-demo-storageclass
+
+Volume Deletion CLI
+-------------------
```
skip a line

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+Volume Deletion CLI
+-------------------
+This section highlights how to delete a volume object in a MetalK8s cluster
```
`s/volume/**Volume**/`

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+This section highlights how to delete a volume object in a MetalK8s cluster
+using the **CLI**
+
+#. Delete a volume.
```
remove the `.` at the end (no period in a title)

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+     volume.storage.metalk8s.scality.com "demo-volume" deleted
+
+
+#. Verify that the volume was deleted.
```
Check that the **Volume** was deleted

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+-------------------
+This section highlights how to delete a volume object in a MetalK8s cluster
+using the **CLI**
+
```
We should have a little explanation somewhere saying that a **Volume** will only be deleted if:
- there is no backing storage
- the volume is not in use

Otherwise, the volume will simply be marked for deletion and remain available until one of the previous condition is met.

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,72 @@
+
+   .. code-block:: shell
+
+     root@bootstrap $ kubectl delete volume <volume_name>
```
```
     root@bootstrap $ kubectl delete volume <volume_name>
     volume.storage.metalk8s.scality.com "<volume_name>" deleted
```

slaperche-scality reviewed (changes requested):
Second commit reviewed

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+Volume Workflow GUI Operations
```
Volume Management using the UI

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+Volume Workflow GUI Operations
+==============================
```
skip a line

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+Volume Creation GUI
+-------------------
+
+.. important::
```
Same remark as for CLI: this is not required, only useful if people want to create their own so can be mentioned before (as I suggested for CLI)

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+Volume Workflow GUI Operations
+==============================
+This section highlights the creation and deletion of MetalK8s volume objects
```
```
This section highlights the creation and deletion of MetalK8s volume objects
using the MetalK8s GUI.
I you want to create a **StorageClass**, you have to use the CLI (see <insert here the reference to the good section>)

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+This section highlights the creation and deletion of MetalK8s volume objects
+using the MetalK8s GUI.
+
+Volume Creation GUI
```
Volume Creation

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+    .. image:: /operation/volume_workflow/img/volume_created.png
+
+
+Volume Deletion GUI
```
Volume Deletion

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+Volume Deletion GUI
+-------------------
+
+.. note::
```
I'm not sure we need a note about that, especially since the UI already warn about it.

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+   is deleted, you will not be able to access it any longer.
+
+To delete a volume from the MetalK8s GUI, from the volume listing, click the
+the delete button.
```
`the the` => remove one

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+
+#. Navigate to the **Nodes** list page, by clicking the button in the sidebar:
+
+    .. image:: /operation/volume_workflow/img/node_list.png
```
You forgot to `git add` the screenshots…

slaperche-scality commented on docs/operation/volume_workflow/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,62 @@
+
+    .. image:: /operation/volume_workflow/img/test_volume.png
+
+#. You should have a new volume enlisted in the **Volume list**.
```
s/enlisted/listed/

slaperche-scality reviewed (dismissed):
Review done

Ebaneck committed:
docs: Rename volume_workflow to volume_management

Ebaneck committed:
docs: Add volume management UI images

MonPote commented on docs/operation/volume_management/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+   Set the following fields:
+
+      - **mountOptions** to specify how the volume should be mounted.
```
Is it possible to provide possible value for `fsType` ? Unlike other field there is not obvious value I can put in IMO.

MonPote commented on docs/operation/volume_management/storageclass_creation.rst:
```diff
@@ -0,0 +1,49 @@
+
+   You can define a new **StorageClass** using the following template:
+
+   .. code-block:: shell
```
I guess `code-block:: yaml` will be better for this case ?

MonPote commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+     volume.storage.metalk8s.scality.com <volume_name> deleted
+
+
+#. Check that the Volume was deleted
```
`has been deleted instead` of `was deleted` ?

MonPote commented on docs/operation/volume_management/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,56 @@
+
+    .. image:: /operation/volume_management/img/node_list.png
+
+#. From the Node list, select the node you would like to create
```
I guess we need a `period` or a `colon` at the end.

Ebaneck replied:
Looks like we are not allowed to use `period` in a list.

I have had to check this too after some review from @slaperche-scality 

FYI: https://data.grammarbook.com/blog/colons/colons/

MonPote commented on docs/operation/volume_management/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,56 @@
+
+    .. image:: /operation/volume_management/img/volume_detail.png
+
+    - Name: Denotes the volume name.
```
Should we put the field in bold like in node expansion ?

MonPote commented on docs/operation/volume_management/volume_creation_deletion_gui.rst:
```diff
@@ -0,0 +1,56 @@
+Volume Deletion
+---------------
+
+To delete a volume from the MetalK8s UI, from the volume listing, click the
```
Do we want to add the modal screenshot  when you click on the trash button ?

Ebaneck replied:
This has been added
https://github.com/scality/metalk8s/pull/1591/files#diff-eddca2048564bd3de81737a3f78077eeR60

MonPote reviewed (dismissed):
LGTM, for a first version. Let's merge it :)

Ebaneck commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/2821b7b6c2fef355959480ec99bf1ec3d73ef1ae) did not succeed in branch feature/1419-document-volume-workflow.




*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality










*The following options are set:* **approve**

Ebaneck committed:
CI: Workaround artefact URL limits

Looks like artefact only supports a limited number of URL. Since this PR
adds metalk8s documentation and images, we uncover the limits of artefact.

To workaround the URL limit, we have removed the wild card URL list limiting it
to just 2 url.

jbertran commented on docs/operation/volume_management/storageclass_creation.rst:
```diff
@@ -0,0 +1,51 @@
+StorageClass Creation
+=====================
+
+**MetalK8s** uses **StorageClass** to configure how **Volumes** are formatted
```
```suggestion
**MetalK8s** uses **StorageClass** objects to describe how **Volumes** are formatted
```

jbertran commented on docs/operation/volume_management/storageclass_creation.rst:
```diff
@@ -0,0 +1,51 @@
+         - rw
+       parameters:
+         fsType: <filesystem_type>
+         mkfsOptions: '["-m", "0"]'
```
Aren't these options also FS-specific? For example, that `-m` is the percentage of reserved blocks in ext4, but some kind of message file in FAT.

slaperche-scality replied:
Yes they are, we should probably make them "generic" as well, and also mention that the field is optional.

Ebaneck replied:
Yes they seem to be `File system` specific but we only support `XFS` and `EXT4` mainly so this would still work i guess. :)

slaperche-scality replied:
Actually no :stuck_out_tongue: 

We already have a wide difference in how to specify `ext4` and `xfs` option.
For instance, even setting the UUID:
- `-U <UUID>` for `ext4`
- `-m uuid=<UUID>` for `xfs`

So `-m 0` isn't gonna work for `xfs` (for the record, reserved blocks are not handled the same way in XFS and ext4, see [here](https://www.spinics.net/lists/xfs/msg30770.html)).

So we definitely wants something generic here, such as `<mkfs_options>`.

jbertran commented on docs/operation/volume_management/storageclass_creation.rst:
```diff
@@ -0,0 +1,51 @@
+      - **mountOptions**: specifies how the volume should be mounted. For
+        example **rw** (read/write), **ro** (read-only).
+      - **fsType**: specifies the filesystem to use on the volume. For example
+        **XFS**, **ext4**.
```
Do we already have documentation on the currently supported FS? If not, this is a good candidate for it.

slaperche-scality replied:
should be `xfs` BTW, we are case-sensitive.

And what do you mean about "documentation on the supported FS"? What could/should we say more than "we support Foo and Bar"?

Ebaneck replied:
We only support this two for now

Ebaneck replied:
The phrase now reads

```
- **fsType**: specifies the filesystem to use on the volume.
        **XFS** and **ext4** are the only currently supported file system types.```

slaperche-scality replied:
Please use `xfs` instead of `XFS`.

Even if the correct typo is indeed `XFS` (uppercase because it's an acronym), the value you can put in the YAML is lowercase only.

jbertran commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+Volume Management using the CLI
+===============================
+
+To use persistent storage in a MetalK8s cluster, one needs to create Volume
```
Let's be consistent in our highlighting of keywords (StorageClass, Volume, ...) with respect to the StorageClass documentation just above. Also, shouldn't these be `terms` leading to glossary entries?

Ebaneck replied:
Yes, they could be terms leading to the glossary entries. But we do not have glossary terms for the following above mentioned words.

Maybe we could track this in another issue

jbertran commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+Volume Creation
+---------------
+
+This section describes how to create a **Volume** from the **CLI**.
```
More specifically, a `Volume` spanning an entire block device.

Ebaneck replied:
Could be controversial since we support `spareLoopDevice` and `rawBlockDevice`

let us keep it simple.

jbertran commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+   - **name**: the name of your volume, must be unique
+   - **nodeName**: the name of the node where the volume will be located.
+   - **storageClassName**: the **StorageClass** to use
+   - **devicePath**: path to the device (on the node).
```
```suggestion
   - **devicePath**: path to the block device (for example, `/dev/sda1`).
```

jbertran commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+
+  .. note:
+
+     A **Volume** can only be deleted if:
```
```suggestion
     A **Volume** object can only be deleted if:
```

jbertran commented on docs/operation/volume_management/volume_creation_deletion_cli.rst:
```diff
@@ -0,0 +1,83 @@
+   The command below will return a list of all volumes. The deleted volume entry
+   should not be found in the list.
+
+   .. code-block:: shell
```
Optional: to me this doesn't render well, I'd rather have the shell block below the note than inside it.

Ebaneck replied:
I would disagree!! Seems to render correctly here: https://eve.devsca.com/github/scality/metalk8s/artifacts/builds/github:scality:metalk8s:staging-2.4.0.r190903163906.c637d72.pre-merge.00010114/html/operation/volume_management/volume_creation_deletion_cli.html#volume-creation

jbertran replied:
Sorry I'm bad at expressing myself... My point it that having the shell block inside the note is not accurate - the note is telling you what to expect, but the shell command should not be part of it since it is what you should run.

Ebaneck replied:
Done!!

jbertran reviewed (commented):
Mostly questions relevant to the amount of work we want to do here (glossary, description of current support, ...).

Well written!

jbertran reviewed (approved):
LGTM!

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye ebaneck.
